pandas/core/arrays/base.py

```diff
@@ -802,12 +809,20 @@ def fillna(
                 npvalues = self.astype(object)
                 func(npvalues, limit=limit, mask=mask)
                 new_values = self._from_sequence(npvalues, dtype=self.dtype)
+                if not copy:
```

Comment:
I don't like this, this is actually slower than ``copy=True``

Reply:
agreed this is not good.  would you prefer to just ignore copy in this case?

Reply:
Yep